src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1260,38 +1260,75 @@ protected function prepare_item_for_database( $request ) {
 	 * @param WP_Post_Type $post_type   Post type.
 	 * @return string|WP_Error Post status or WP_Error if lacking the proper permission.
 	 */
-	protected function handle_status_param( $post_status, $post_type ) {
-
-		switch ( $post_status ) {
-			case 'draft':
-			case 'pending':
-				break;
-			case 'private':
-				if ( ! current_user_can( $post_type->cap->publish_posts ) ) {
-					return new WP_Error(
-						'rest_cannot_publish',
-						__( 'Sorry, you are not allowed to create private posts in this post type.' ),
-						array( 'status' => rest_authorization_required_code() )
-					);
-				}
-				break;
-			case 'publish':
-			case 'future':
-				if ( ! current_user_can( $post_type->cap->publish_posts ) ) {
-					return new WP_Error(
-						'rest_cannot_publish',
-						__( 'Sorry, you are not allowed to publish posts in this post type.' ),
-						array( 'status' => rest_authorization_required_code() )
-					);
-				}
-				break;
-			default:
-				if ( ! get_post_status_object( $post_status ) ) {
-					$post_status = 'draft';
-				}
-				break;
+	protected function handle_status_param( $post_status, $post_type, $post_id = null ) {
```

Comment:
Let's default this to `0` so we can keep a single type of `integer` for the doc block. We'll also need to add that parameter with `@param` to the doc block. We should also add a `@since` entry describing that we've made a change. You can take a look at an example in `rest_validate_value_from_schema`.

Reply:
@TimothyBJacobs  Which Version should be used for the `@since `entry?

Reply:
`5.6.0`.